specs/discussions/2025-05-segments-global-history.md

```diff
@@ -0,0 +1,320 @@
+   final, by their corresponding intermediate shard block reaching the
+   `FINALITY_PROBABILITY_THRESHOLD`, these segments and any beacon chain segments included in the
+   block are added to the global history, and a super segment is created from the segments in the
+   block. The super segment is created by computing the Merkle root of the tree of segments
+   aggregated in the super segment, and is included in the `super_segment_root` field of the
+   `SuperSegmentHeader`.
+   - To compute the super segment root, segments are ordered starting from the `own_segments_root`
+     of the beacon chain, and then in increasing order according to their shard ID and the
+     underlying block height in which the segments (or their confirmation) where included (e.g.
+     `own_segment_info_1`, `own_segment_info_2`, `intermediate_shard_segment_shard1`,
+     child_shard_segment_shard1`, `intermediate_shard_segment_shard2`, etc.).
+   - The super segment index is computed is determined by the number of segments aggregated in the
+     super segment, and the index of the super segment in the global history of the system. If the
+     previous super segment has `super_segment_index` 4 as it aggregates 4 super segments, if the
+     next super segments aggregates 5 segments, the `super_segment_index` of the next super segment
+     will be 4 + 5 = 9. In this way, the global history index for every segment of the shard is
+     assigned.
+
+```rust
+struct SuperSegmentHeader {
+	// Root hash of the previous super segment that chains all super segments together.
+	prev_super_segment_root: Hash,
+	// The root of the super segment.
+	// It is computed by getting the Merkle root of the tree of segments aggregated in the super segment.
+	super_segment_root: Hash,
+	// Number of segments aggregated in this super segment.
+   // TODO: Num segments not needed because we can look at the block in the beacon height to get the
+   // number of segments, their shards and all the actual information.
+	num_segments: u64
+	// Index of the super segment in the global history of the system
+	// (e.g. if the previous segment had super_segment_index 0, and num_segments 4,
+	// this super segment will have super_segment_index 4).
+	super_segment_index: u64,
+	// Beacon height in which the super segment was created. This is useful to inspect the block
+	// for additional information about the transactions with segment creations
+	beacon_height: BlockNumber,
+}
+```
+
+8. Farmers in lower-level shards are following the beacon chain and periodically checking for new
+   super segment creations. As soon as a new super segment is created, the farmer entitled to create
+   the next block will include a transaction that notifies the chain about the new super segment and
+   the list of segments from the shard that are part of it and can be conveniently sealed into a
+   `VerifiedSegment` as it has been included in the global history. The inclusion of a new
```

Comment:
It is not checking anything periodically. Following the chain implies processing of blocks and seeing super segment right when they are created, not at some other interval.

I'd re-write the first sentence like this:
```suggestion
8. Farmers in lower-level shards are following the beacon chain and as soon as a new super segment is created, the next block producer includes information about it in the block.
```

However, I don't understand why it is needed to include either super segment or verified segment in the block at all. Could you elaborate its purpose?